nova/tests/unit/scheduler/test_external_scheduler_filter.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2024 SAP SE or an SAP affiliate company.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+# Tests for external scheduler filter.
+
+from unittest.mock import MagicMock
+from unittest.mock import patch
+
+import nova.conf
+from nova.scheduler.filters import external_scheduler_filter
+from nova import test
+from nova.tests.unit.scheduler import fakes
+
+
+# Set this the api url to something not empty,
+# so the scheduler is invoked but mock the actual communication.
+CONF = nova.conf.CONF
+CONF.filter_scheduler.external_scheduler_api_url = "http://127.0.0.1:1234"
```

Comment:
Not entirely comfortable with setting a value on import even if it is just a test, I'd suggest to add:
```python
class ExternalSchedulerFilterTestCase(test.NoDBTestCase):
  def setUp(self):
     super(test.NoDBTestCase, self).setUp()
     CONF.set_override('external_scheduler_api_url', 'http://127.0.0.1:1234', 'filter_scheduler')
```

Reply:
done